### PR TITLE
Fix deployment to maven central.

### PR DESCRIPTION
**Pull request by rozza**
- Pom files were missing their descriptions so failed the sonatype maven central check
- Scala docs were wrongly classified so failed the sonatype maven central check
- Scala 2.13 didn't produce scaladocs at all - workaround found in https://github.com/gradle/gradle/issues/9855 and included the missing scala-ant jar that was removed in Scala 2.13

### CODE REVIEW

**Comment by rozza** [comment]
I created `gradle/scala/lib/scala-ant-2.13.1.jar ` by building https://github.com/cigaly/scala-ant which is a port of the scala code from 2.12 that was removed in 2.13.

File: gradle/publish.gradle [hunk 1]
```diff
@@ -123,7 +123,12 @@ configure(scalaProjects) { project ->
 
     task scaladocJar(type: Jar) {
         from scaladoc
-        classifier = 'scaladoc'
+        classifier = 'javadoc'
+
+        // Workaround for gradle 2.13 scaladoc issue: https://github.com/gradle/gradle/issues/9855
+        if (project.scalaVersion.startsWith("2.13")) {
+            project.scaladoc.scalaClasspath += files("$rootDir/gradle/scala/lib/scala-ant-2.13.1.jar")
```
Comment (jyemin) [inline gradle/publish.gradle]: How does this file get to this location?

File: gradle/publish.gradle [hunk 2]
```diff
@@ -144,6 +146,8 @@ configure(scalaProjects) { project ->
 
     afterEvaluate {
         jar configureJarManifestAttributes(project)
+        publishing.publications.mavenJava.artifactId = project.archivesBaseName
```
Comment (jyemin) [inline gradle/publish.gradle]: Does this result in a change to the POM? Or is it for consistency with Java section?

File: gradle/publish.gradle (hunk 1 above)
Comment (rozza) [reply gradle/publish.gradle]: I compiled it and added it to this location.

File: gradle/publish.gradle (hunk 2 above)
Comment (rozza) [reply gradle/publish.gradle]: For consistency with the Java section

File: gradle/publish.gradle (hunk 1 above)
Comment (rozza) [reply gradle/publish.gradle]: Source code for the jar: https://github.com/cigaly/scala-ant

Comment (jyemin) [reply gradle/publish.gradle]: Oh, I totally missed the new file in the PR because it's binary

**Pull request closed without merging**